src/content/en/fundamentals/media/fast-playback-with-video-preload.md

```diff
@@ -0,0 +1,571 @@
+segment files. It exists some middleware libraries such as [Google's Shaka
+Player], [JW Player], and [Video.js] dedicated to this.
+
+```
+<video id="video" controls></video>
+
+<script>
+  const mediaSource = new MediaSource();
+  video.src = URL.createObjectURL(mediaSource);
+  mediaSource.addEventListener('sourceopen', sourceOpen);
+
+  function sourceOpen() {
+    URL.revokeObjectURL(video.src);
+    const sourceBuffer = mediaSource.addSourceBuffer('video/webm; codecs="vp9"');
+
+    // Fetch beginning of the video by setting the Range HTTP request header.
+    fetch('file.webm', { headers: { range: 'bytes=0-567139' } })
+    .then(response => response.arrayBuffer())
+    .then(data => {
+      sourceBuffer.appendBuffer(data);
+      sourceBuffer.addEventListener('updateend', updateEnd, { once: true });
+    });
+  }
+
+  function updateEnd() {
+    // Video is now ready to play!
+    var bufferedSeconds = video.buffered.end(0) - video.buffered.start(0);
+    console.log(bufferedSeconds + ' seconds of video are ready to play!');
+
+    // Fetch the rest of the video when user starts playing video.
+    video.addEventListener('playing', fetchNextSegment, { once: true });
+  }
+
+  function fetchNextSegment() {
+    fetch('file.webm', { headers: { range: 'bytes=567140-1196488' } })
+    .then(response => response.arrayBuffer())
+    .then(data => {
+      const sourceBuffer = mediaSource.sourceBuffers[0];
+      sourceBuffer.appendBuffer(data);
+    });
+  }
+</script>
+```
+
+### Battery awareness
+
+I would suggest you take into account the battery level of user's device before
+thinking about preloading a video in order to preserve battery life when the
+power level is low.
+
+Please disable preload or at least preload a lower resolution video when the
+device is running out of battery.
+
+
+```
+if ('getBattery' in navigator) {
+  navigator.getBattery()
+  .then(battery => {
+    // If battery is charging or battery level is high enough
+    if (battery.charging || battery.level > 0.15) {
+      // TODO: Preload the first segment of a video.
+    } else {
+      // TODO: Enable some kind of Power Saving mode.
+    }
+  });
+} else {
+  // Battery Status API is not supported.
+  // TODO: Preload the first segment of a video.
+}
+```
+
+
+### Pre-cache multiple first segments
+
+Now, what if I want to speculatively pre-load some media content without
+knowing which piece of media user will eventually pick. If user is on a web
+page that contains 10 videos, we probably have enough memory to fetch one
+segment file from each but we should definitely not create 10 hidden video
+elements and 10 `MediaSource` objects and start feeding that data.
+
+The 2-parts example below shows you how to pre-cache multiple first segments of
+video using the powerful and easy-to-use Cache API. Note that something similar
+can be achieved with IndexedDB as well. We're not using service workers yet as
+the Cache API is also accessible from the Window object.
```

Comment:
It is important as developers might not know the Cache API (always used with Service Worker) can also be used without Service Worker, from the Window object.

That makes me think I should replace `caches` with `window.caches` in code snippets below.